core/src/validator.rs

```diff
@@ -69,15 +68,15 @@ impl Validator {
     /// - Validator denied the operation
     pub fn validate(
         &self,
-        wsv: &WorldStateView,
+        wsv: &mut WorldStateView,
         authority: &AccountId,
         operation: impl Into<data_model_validator::NeedsValidationBox>,
     ) -> Result<()> {
         let operation = operation.into();
 
         let runtime = wasm::RuntimeBuilder::new()
-            .with_engine(self.engine.clone()) // Cloning engine is cheap, see [`wasmtime::Engine`] docs
-            .with_configuration(wsv.config.borrow().wasm_runtime_config)
+            .with_engine(wsv.engine.clone()) // Cloning engine is cheap, see [`wasmtime::Engine`] docs
```

Comment:
So now we have the same engine for smart contracts, triggers and validators? Will hope it works

Reply:
I think as long as we use the same global configuration everything should be fine.

> Engines store global configuration preferences such as compilation settings, enabled features, etc. You’ll likely only need at most one of these for a program.

From wasmtime docs.

Reply:
@Arjentix so i think we can reintroduce second engine if for example we decide to turn off fuel for validator.